onnxruntime/core/framework/provider_bridge_ort.cc

```diff
@@ -1157,3 +1187,83 @@ ORT_API_STATUS_IMPL(OrtApis::SessionOptionsAppendExecutionProvider_CUDA, _In_ Or
   options->provider_factories.push_back(factory);
   return nullptr;
 }
+
+ORT_API_STATUS_IMPL(OrtApis::CreateTensorRTProviderOptions, _Outptr_ OrtTensorRTProviderOptions** out) {
+#ifdef USE_TENSORRT
+  *out = new OrtTensorRTProviderOptions();
+  (*out)->trt_int8_calibration_table_name = nullptr;
+  (*out)->trt_engine_cache_path = nullptr;
+  (*out)->trt_engine_decryption_lib_path = nullptr;
+  return nullptr;
+#else
+  ORT_UNUSED_PARAMETER(out);
+  return CreateStatus(ORT_FAIL, "TensorRT execution provider is not enabled in this build.");
+#endif
+}
+
+ORT_API_STATUS_IMPL(OrtApis::UpdateTensorRTProviderOptions,
```

Comment:
are these the implementations of the C API functions?
if so, should we ensure that exceptions do not propagate outside of these calls (e.g., with API_IMPL_BEGIN/API_IMPL_END)? and why are these C API functions defined in provider_bridge_ort.cc?

Reply:
I was wondering what kind of functions should be in the C API?
For these four functions, it's related to TRT EP functionality. So I think I will move the definition of them out of C API and define/implemented in provider_bridge_ort.cc and tensorrt_provider_factor.h

Reply:
Sorry, looks like you had to go back & forth with these functions & the C API, but this is good. The only issue I see is if they need API_IMPL_BEGIN/API_IMPL_END like Edward mentions above, just to be sure we don't throw any exceptions through them.

Reply:
Yes, API_IMPL_BEGIN/API_IMPL_END is needed. Just added them.